seqr/views/apis/data_manager_api.py

```diff
@@ -72,26 +93,27 @@ def elasticsearch_status(request):
         for index_prefix in list(seqr_index_projects.keys()):
             if index_name.startswith(index_prefix):
                 projects_for_index += list(seqr_index_projects.pop(index_prefix).keys())
-        index['projects'] = [{'projectGuid': project.guid, 'projectName': project.name} for project in projects_for_index]
+        index['projects'] = [
+            {'projectGuid': project.guid, 'projectName': project.name} for project in projects_for_index]
 
-    errors = ['{} does not exist and is used by project(s) {}'.format(
-        index, ', '.join(['{} ({} samples)'.format(p.name, len(indivs)) for p, indivs in project_individuals.items()])
-    ) for index, project_individuals in seqr_index_projects.items() if project_individuals]
+    return indices, seqr_index_projects
 
-    return create_json_response({
-        'indices': indices,
-        'diskStats': list(disk_status.values()),
-        'elasticsearchHost': ELASTICSEARCH_SERVER,
-        'errors': errors,
-    })
 
+@data_manager_required
+def delete_index(request):
+    index = json.loads(request.body)['index']
+    active_index_samples = Sample.objects.filter(is_active=True, elasticsearch_index=index)
+    if active_index_samples:
+        projects = {
+            sample.individual.family.project.name for sample in active_index_samples.select_related('individual__family__project')
+        }
+        return create_json_response({'error': 'Index "{}" is still used by: {}'.format(index, ', '.join(projects))}, status=403)
```

Comment:
Is there a way that the users can delete the samples that using a specific index? In some cases, users may want to replace an old index with a new one. To delete the old index, the users have to delete the dataset for the old index.

Reply:
The samples are records of what indices ever existed and are never deleted. They are marked inactive when they are no longer used. There is a different user interface for doing that, this is ONLY for deleting unused indices from elasticsearch